src/shared/vscode/env.ts

```diff
@@ -97,17 +97,42 @@ type ServiceConfig = Partial<{ [K in keyof ConfigToEnvMap]: any }>
 const logConfigsOnce: { [key: string]: ReturnType<typeof onceChanged> } = {}
 
 /**
- * Accepts a service name and a {config key -> expected env var name} map.
+ * Generate a map of environment variable names to environment variables.
+ *
+ * It does this by converting camel case entries in envVarNames to uppercase joined by underscores
+ * If there is no camel case in an entry in envVarNames then it just converts the envVar to uppercase
+ *
+ * E.g. if service is codecatalyst:
+ *  gitHostname -> __CODECATALYST_GIT_HOSTNAME
+ *  region -> __CODECATALYST_REGION
+ */
+function getEnvVars(service: string, envVarNames: string[]) {
+    const envVars = new Map<string, string>()
+    for (const name of envVarNames) {
+        // convert camel case to uppercase joined by underscores
+        // e.g. gitHostname -> GIT_HOSTNAME
+        const envVarName = name
+            .split(/(?=[A-Z])/)
+            .map(s => s.toUpperCase())
+            .join('_')
+        envVars.set(name, `__${service.toUpperCase()}_${envVarName}`)
+    }
+    return envVars
+}
+
+/**
+ * Accepts a service name and an array of configs.
  * For each config key, check if the associated env var exists and return
  * a config map with the found values. Changes are logged once for each
  * service/found env var combos.
  */
-export function getServiceEnvVarConfig(service: string, configToEnvMap: ConfigToEnvMap): ServiceConfig {
+export function getServiceEnvVarConfig(service: string, configs: string[]): ServiceConfig {
```

Comment:
let's remove the ConfigToEnvMap type, and update the ServiceConfig type on lines 95,96